packages/datatrak-web-server/src/routes/SubmitSurveyReponse/handleTaskCompletion.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+import { QUERY_CONJUNCTIONS } from '@tupaia/database';
+import { DatatrakWebServerModelRegistry } from '../../types';
+import { MeditrakSurveyResponseRequest } from '@tupaia/types';
+
+export const handleTaskCompletion = async (
+  models: DatatrakWebServerModelRegistry,
+  surveyResponse: MeditrakSurveyResponseRequest,
+) => {
+  const {
+    id: surveyResponseId,
+    survey_id: surveyId,
+    entity_id: entityId,
+    data_time: dataTime,
+    user_id: userId,
+  } = surveyResponse;
+  const tasksToComplete = await models.task.find({
```

Comment:
This query is almost a copy of the query that is in the TaskCompletionHandler but it is for searching based on 1 survey response instead of many. I had a try at bringing them together but it was becoming more complicated than I liked. I'm open to suggestions about a nice way of doing that.

Reply:
The only way I can think of is maybe adding something like `getAssociatedTasks` to the `SurveyResponse` model but even that I'm not sure about